[jsk_2016_01_baxter_apc] rbo callback error with SoftKinetic Camera

@yuyu2172 now #1410 is fixed, and i tried to use softkinetic on both arm, but i got these errors.

```
[INFO] [WallTime: 1462887464.394961] started
[INFO] [WallTime: 1462887465.471998] get_spatial elapsed time 0.999156951904
[ERROR] [WallTime: 1462887465.559949] bad callback: <bound method RBOSegmentationInBinNode._callback of <__main__.RBOSegmentationInBinNode object at 0x7f8308afb1d0>>
Traceback (most recent call last):
  File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
    cb(msg)
  File "/home/baxter/ros/indigo/src/start-jsk/jsk_apc/jsk_2016_01_baxter_apc/node_scripts/rbo_segmentation_in_bin_node.py", line 85, in _callback
    self.segmentation()
  File "/home/baxter/ros/indigo/src/start-jsk/jsk_apc/jsk_apc2016_common/python/jsk_apcIndexError: index 299 is out of bounds for axis 1 with size 1542016_common/segmentation_in_bin/rbo_segmentation_in_bin.py", line 57, in segmentation
    desired_object=self.target_object)
  File "/home/baxter/ros/indigo/src/start-jsk/jsk_apc/jsk_apc2016_common/python/jsk_apc2016_common/rbo_segmentation/probabilistic_segmentation.py", line 603, in predict
    self.backproject()
  File "/home/baxter/ros/indigo/src/start-jsk/jsk_apc/jsk_apc2016_common/python/jsk_apc2016_common/rbo_segmentation/probabilistic_segmentation.py", line 563, in backproject
    self.likelihood_images[feature_name][object_name][self.bin_mask] = Utils.backproject(self.likelihoods[feature_name][object_name], self.feature_images[feature_name][self.bin_mask])
IndexError: index 299 is out of bounds for axis 1 with size 154
```k_apc2016_common/segmentation_in_bin/rbo_segmentation_in_bin.py", line 57, in segmentation
    desired_object=self.target_object)
  File "/home/baxter/ros/indigo/src/start-jsk/jsk_apc/jsk_apc2016_common/python/jsk_apc2016_common/rbo_segmentation/probabilistic_segmentation.py", line 603, in predict
    self.backproject()
  File "/home/baxter/ros/indigo/src/start-jsk/jsk_apc/jsk_apc2016_common/python/jsk_apc2016_common/rbo_segmentation/probabilistic_segmentation.py", line 563, in backproject
    self.likelihood_images[feature_name][object_name][self.bin_mask] = Utils.backproject(self.likelihoods[feature_name][object_name], self.feature_images[feature_name][self.bin_mask])
IndexError: index 299 is out of bounds for axis 1 with size 154
```

i thought it's because of size difference between point cloud and rgb image, so i resize rgb image color to same size as point cloud.
however, i got same error, but the last line is different as below.

```
IndexError: index 217 is out of bounds for axis 0 with size 217
```

@yuyu2172 
upper: rgb image color shape
lower: mask color shape

```
[INFO] [WallTime: 1463062655.533600] started
[INFO] [WallTime: 1463062656.655322] get_spatial elapsed time 0.991035938263
(467, 469)
(572, 648)
/home/baxter/ros/indigo/src/start-jsk/jsk_apc/jsk_apc2016_common/python/jsk_apc2016_common/rbo_segmentation/probabilistic_segmentation.py:565: VisibleDeprecationWarning: boolean index did not match indexed array along dimension 0; dimension is 467 but corresponding boolean dimension is 572
  self.likelihood_images[feature_name][object_name][self.bin_mask] = Utils.backproject(self.likelihoods[feature_name][object_name], self.feature_images[feature_name][self.bin_mask])
[ERROR] [WallTime: 1463062656.733234] bad callback: <bound method RBOSegmentationInBinNode._callback of <__main__.RBOSegmentationInBinNode object at 0x7ff66a8dc210>>
Traceback (most recent call last):
  File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
    cb(msg)
  File "/home/baxter/ros/indigo/src/start-jsk/jsk_apc/jsk_2016_01_baxter_apc/node_scripts/rbo_segmentation_in_bin_node.py", line 85, in _callback
    self.segmentation()
  File "/home/baxter/ros/indigo/src/start-jsk/jsk_apc/jsk_apc2016_common/python/jsk_apc2016_common/segmentation_in_bin/rbo_segmentation_in_bin.py", line 57, in segmentation
    desired_object=self.target_object)
  File "/home/baxter/ros/indigo/src/start-jsk/jsk_apc/jsk_apc2016_common/python/jsk_apc2016_common/rbo_segmentation/probabilistic_segmentation.py", line 605, in predict
    self.backproject()
  File "/home/baxter/ros/indigo/src/start-jsk/jsk_apc/jsk_apc2016_common/python/jsk_apc2016_common/rbo_segmentation/probabilistic_segmentation.py", line 565, in backproject
    self.likelihood_images[feature_name][object_name][self.bin_mask] = Utils.backproject(self.likelihoods[feature_name][object_name], self.feature_images[feature_name][self.bin_mask])
IndexError: index 467 is out of bounds for axis 0 with size 467
```

--- a/jsk_2016_01_baxter_apc/launch/include/segmentation_for_each_hand.launch
+++ b/jsk_2016_01_baxter_apc/launch/include/segmentation_for_each_hand.launch
@@ -49,7 +49,7 @@
   <node name="$(arg HAND)_sib_mask_image_to_pi"
         pkg="nodelet" type="nodelet"
         args="load jsk_pcl/MaskImageToPointIndices $(arg MANAGER)">
-    <remap from="~input" to="$(arg HAND)_sib_rbo/target_mask"/>
+    <remap from="~input" to="$(arg HAND)_sib_rbo/target_mask_compressed"/>
   </node>
 
   <node name="$(arg HAND)_sib_extract_pi"

--- a/jsk_apc2016_common/node_scripts/rbo_segmentation_in_bin_node.py
+++ b/jsk_apc2016_common/node_scripts/rbo_segmentation_in_bin_node.py
@@ -30,6 +30,7 @@ class RBOSegmentationInBinNode(ConnectionBasedTransport):
 
         self.bridge = CvBridge()
         self.img_pub = self.advertise('~target_mask', Image, queue_size=100)
+        self.img_compressed_pub = self.advertise('~target_mask_compressed', Image, queue_size=100)
         self.posterior_pub = self.advertise('~posterior', Image, queue_size=20)
         self.masked_input_img_pub = self.advertise('~masked_input', Image, queue_size=20)
 
@@ -79,11 +80,16 @@ class RBOSegmentationInBinNode(ConnectionBasedTransport):
         self.set_apc_sample()
         # generate a binary image
         self.segmentation()
+        predicted_segment_compressed = cv2.resize(self.predicted_segment, (dist_msg.width, dist_msg.height))
         try:
             predict_msg = self.bridge.cv2_to_imgmsg(
                     self.predicted_segment, encoding="mono8")
+            predict_compressed_msg = self.bridge.cv2_to_imgmsg(
+                    predicted_segment_compressed, encoding="mono8")
             predict_msg.header = color_msg.header
+            predict_compressed_msg.header = color_msg.header
             self.img_pub.publish(predict_msg)
+            self.img_compressed_pub.publish(predict_compressed_msg)
         except CvBridgeError as e:
             rospy.logerr('{}'.format(e))
 

